system/src/system_connection_manager.h

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2023 Particle Industries, Inc.  All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef SYSTEM_CONNECTION_MANAGER_H
+#define SYSTEM_CONNECTION_MANAGER_H
+
+#include "hal_platform.h"
+
+#if HAL_PLATFORM_IFAPI
```

Comment:
I think `ConnectionTester` class is heavily reliant on the socket api and should be included with `#if HAL_PLATFORM_IFAPI`, but should `ConnectionManager` be available to all platforms?

Reply:
I think we should add a PLATFORM_ define for this and enable it only on MSOM platform for now.

Reply:
We are using this now as a replacement for the NTP-based internet test, so this should be enabled on all platforms, not only for ACM purposes.